src/main/java/org/thoughtcrime/securesms/connect/DcEventCenter.java

```diff
@@ -177,6 +178,10 @@ public long handleEvent(@NonNull DcEvent event) {
         DcHelper.getNotificationCenter(context).removeNotifications(accountId, event.getData1Int());
         break;
 
+      case DcContext.DC_EVENT_ACCOUNTS_BACKGROUND_FETCH_DONE:
+        FetchForegroundService.stop(context);
```

Comment:
imu, same as on iOS, we need to wait for `DC_EVENT_ACCOUNTS_BACKGROUND_FETCH_DONE` to make sure all events are processed - otherwise, the app may be terminated just after `backgroundFetch()`, leaving `DC_EVENT_INCOMING_MSG` unprocessed, resulting in no notification being shown